src/app/clusters/camera-av-stream-management-server/camera-av-stream-management-server.cpp

```diff
@@ -1001,15 +1001,22 @@ void CameraAVStreamMgmtServer::ModifyVideoStream(const uint16_t streamID, const
     {
         if (stream.videoStreamID == streamID)
         {
+            bool wasModified = false;
             if (waterMarkEnabled.HasValue())
             {
+                wasModified = !stream.watermarkEnabled.HasValue() || stream.watermarkEnabled.Value() != waterMarkEnabled.Value();
                 stream.watermarkEnabled = waterMarkEnabled;
             }
             if (osdEnabled.HasValue())
             {
+                wasModified       = wasModified || !stream.OSDEnabled.HasValue() || stream.OSDEnabled.Value() != osdEnabled.Value();
                 stream.OSDEnabled = osdEnabled;
             }
-            ChipLogError(Camera, "Modified video stream with ID: %d", streamID);
+            if (wasModified)
+            {
+                PersistAndNotify<Attributes::AllocatedVideoStreams::Id>();
+            }
+            ChipLogProgress(Camera, "Modified video stream with ID: %d", streamID);
```

Comment:
I know I missed this in the shared patch, but I think we should move this log to the `if(wasModified)` block.

Reply:
I overlooked that too. Applied the change. Thanks